src/SignalR/common/SignalR.Common/src/Protocol/CompletionMessage.cs

```diff
@@ -8,11 +8,11 @@ namespace Microsoft.AspNetCore.SignalR.Protocol
 {
     public class CompletionMessage : HubInvocationMessage
     {
-        public string Error { get; }
-        public object Result { get; }
+        public string? Error { get; }
+        public object? Result { get; }
         public bool HasResult { get; }
 
-        public CompletionMessage(string invocationId, string error, object result, bool hasResult)
+        public CompletionMessage(string? invocationId, string? error, object? result, bool hasResult)
```

Comment:
I don't think invocationId should ever be null for this type

Reply:
I can mark this as non-null. `HubInvocationMessage` allows null values and none of these APIs check for null, so it was difficult to make that call. That said, we've tried to lean towards most common use rather than absolute correctness for most of our APIs because it makes everyone's lives a bit simpler.